src/login.ts

```diff
@@ -0,0 +1,24 @@
+import { compare } from 'bcrypt';
+import { appDataSource } from './data-source';
+import { User } from './entity/user';
+import { LoginInput } from './schema';
+import { BaseError } from './base-error';
+
+export async function getUserFromLoginInput(credentials: LoginInput) {
+  const userRepository = appDataSource.getRepository(User);
+  const user = await userRepository.findOne({ where: { email: credentials.email } });
```

Comment:
```suggestion
  const user = await userRepository.findOneBy({ email: credentials.email });
```

Esse `findOneBy` é um atalho quando vc só usa `where` e não usa outras opções na busca.